### DISCUSSION TITLE
How to press & double press

### DISCUSSION THREAD

**Discussion by gartkb** [AntiMicroX/antimicrox#758] [Q&A]
Hello

The wiki says : Cycle Reset: 0.50 seconds
Release 0.01 -> Pause 0.00 -> Release 0.49 -> E -> Cycle -> Release 0.01 -> R

But when I do this I get unreliable results. Can you clarify? thank you!

**Comment by gartkb** [comment]
Answering my own question, what I came up with below seems to work. But I don't understand it very well so it's kind of trial and error. Maybe if someone comes along that understands they can suggest improvements. But otherwise I think it's not good the wiki has bad info on it.

Release 0.01 -> Pause 0.00 -> Release 0.01 -> E -> Cycle 0.5 -> Release 0.01 -> R

Also putting the cycle time before the series makes it confusing. It should be written as I have above.
Hope this helps somebody :) Or if you have an improvement please comment!

Actually I determined this wasn't working properly. This seems to work, but not very well unfortunately. I'd really love some help with a proper press / double press implementation.

Release .01 -> Pause 0.00 -> Release 0.08 -> E -> Cycle (.5 Auto) -> Release .01 -> R
I think this works, but it ignores very quick single presses. It's not really usable, but you could if you had to.

**Reply by redwolf460** [reply]
I dont understand would this allow me to do something like (xbox controller) Y + B = V (on the key board)

**Comment by mrbeauregarde** [comment]
So, fundamentally, we have

1. Tap (Dn>Up>!Dn)
2. Hold (!Dn:Dn>!Up>Up)
3. No Press (!Dn)
4. Release (Up)
5. Pause (Up>?!Dn/!Any?)
6. Cycle (Dn>Up)
7. Cycle Reset (!Any)
8. Delay (!Up:Dn>

Bear with me, this is still kind of coming to me as I type. 
### Hold (!Dn:Dn>!Up>Up)
Any Zones activated before a Hold Zone are released, hence _**!Dn**_:Dn>!Up>Up . If it's correct that the edge is on press, hence _**:Dn**_ . Because it's a time threshold, hence _**>!Up**_ . I actually think release isn't required in which case _**>Up)**_ is incorrect, standby.

Who cares though right? But we're dealing with an apparatus constructed from lot of fixed form components. It's a lot like a marble track, if the end of one isnt connected to the start of another, the marble isnt going to do interesting, it's just going to fall straight to the bottom. If a zone finishes in Dn, it's gotta come up to go down again. Sometimes that's because we stop pressing, other times its because Hold or Pause "un-downs" them.

I haven't sat down and tested each one to see exactly which release when and such, the fundamentals arent really fundamental nor are they a complete list, and I have concerns the logic for at least a few isn't correct. This is more of a "Hey, if you want to co-op this thing, I'm down" sort of reply and a glimpse into my thinking on it. I'll probably be back soon to edit this to a level of less incorrect. Cheers
